drivers/led/led_gpio.c

```diff
@@ -0,0 +1,261 @@
+/*
+ * Copyright (c) 2019 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#include <drivers/gpio.h>
+#include <drivers/led.h>
+#include <logging/log.h>
+
+LOG_MODULE_REGISTER(led_gpio, CONFIG_LED_GPIO_LOG_LEVEL);
+
+#define LED_SW_BLINK BIT(0)
+#define LED_BLINK_DISABLE BIT(1)
+
+struct gpio_cfg {
+	const char *port;
+	u8_t pin;
+	u8_t polarity;
+};
+
+struct led_dts_cfg {
+	struct gpio_cfg gpio;
+	const char *label;
+	u32_t id;
+};
+
+struct led_gpio_data {
+	struct device *gpio_dev;
+	const struct led_dts_cfg *cfg;
+#if defined(CONFIG_LED_SOFTWARE_BLINK)
+	struct k_timer timer;
+	atomic_t flags;
+	u32_t blink_delay_on;
+	u32_t blink_delay_off;
+	bool next_state;
+#endif
+	bool state;
+};
+
+struct led_gpio_cfg {
+	const struct led_dts_cfg *led;
+	u32_t led_cnt;
+};
+
+static inline struct led_gpio_data *get_dev_data(struct device *dev)
+{
+	return dev->driver_data;
+}
+
+static inline const struct led_gpio_cfg *get_dev_config(struct device *dev)
+{
+	return dev->config->config_info;
+}
+
+static struct led_gpio_data *led_parameters_get(struct device *dev, u32_t led)
+{
+	const struct led_gpio_cfg *cfg = get_dev_config(dev);
+	struct led_gpio_data *data = get_dev_data(dev);
+
+	for (size_t i = 0; i < cfg->led_cnt; i++) {
+		if (data[i].cfg->id == led) {
+			return &data[i];
+		}
+	}
+
+	return NULL;
+}
+
+static int gpio_led_set_no_delay(struct led_gpio_data *data, bool state)
+{
+	int err;
+
+	data->state = state;
+
+	err = gpio_pin_write(data->gpio_dev, data->cfg->gpio.pin,
+		state ? ((data->cfg->gpio.polarity & GPIO_INT_ACTIVE_HIGH) != 0) :
+		((data->cfg->gpio.polarity & GPIO_INT_ACTIVE_HIGH) == 0));
```

Comment:
What about using someting like this:
```
	u32_t pin_value;

	if (data->cfg->gpio.polarity & GPIO_INT_ACTIVE_HIGH) {
		pin_value = (state ? 1 : 0);
	} else {
		pin_value = (state ? 0 : 1);
	}
```
Could that make things a bit clearer (and shorten the lines so that checkpatch no longer complains)?

Reply:
I changed it also in init function.